packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/steps-reducer.ts

```diff
@@ -0,0 +1,45 @@
+import { TSelfie } from '../signup-wizard-steps/selfie-step/selfie-step';
+
+type TStepsState = {
+    selfie: {
+        selfie_with_id: TSelfie;
+    } | null;
+    is_selfie_step_enabled: boolean;
+};
+
+const ACTION_TYPES = {
+    SET_SELFIE: 'SET_SELFIE',
+    SET_SELFIE_STEP_ENABLED: 'SET_SELFIE_STEP_ENABLED',
+} as const;
+
+// Action creators
+export const setSelfie = (value: { selfie_with_id: TSelfie }) => {
+    return {
+        type: ACTION_TYPES.SET_SELFIE,
+        value,
+    };
+};
+
+export const setSelfieStepEnabled = (value: boolean) => {
+    return {
+        type: ACTION_TYPES.SET_SELFIE_STEP_ENABLED,
+        value,
+    };
+};
+
+// Initial state
+export const initial_state = { selfie: null, is_selfie_step_enabled: false };
+
+// Reducer
```

Comment:
@george-usynin-binary We can simplify and isolate this reducer, Let’s have a call and do it in another PR 🙇🏻